git_proxy.go

```diff
@@ -0,0 +1,44 @@
+package main
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"net/http/httputil"
+	"net/url"
+)
+
+type TestProxy struct {
```

Comment:
suggestion: even tough this lives in the main package, I'd try to avoid exporting this test helper. Even moving it to a test file could be of interest. Thoughts?

Reply:
Moved it to a separate package now, is this better?